modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbMvPartitionStorage.java

```diff
@@ -1015,6 +1117,149 @@ public void destroyData(WriteBatch writeBatch) throws RocksDBException {
         writeBatch.deleteRange(cf, partitionStartPrefix(), partitionEndPrefix());
     }
 
+    @Override
+    public @Nullable TableRowAndRowId pollForVacuum(HybridTimestamp lowWatermark) {
+        return busy(() -> {
+            WriteBatchWithIndex batch = requireWriteBatch();
+
+            try (
+                    var gcUpperBound = new Slice(partitionEndPrefix());
+                    ReadOptions gcOpts = new ReadOptions().setIterateUpperBound(gcUpperBound).setTotalOrderSeek(true);
+                    RocksIterator gcIt = db.newIterator(gc, gcOpts);
+            ) {
+                gcIt.seek(partitionStartPrefix());
+
+                if (invalid(gcIt)) {
+                    // GC queue is empty.
+                    return null;
+                }
+
+                ByteBuffer gcKey = allocateDirect(GC_KEY_SIZE);
```

Comment:
Let's use the buffer too, shall we?

Reply:
What do you mean, it is used

Reply:
Let's use `ThreadLocal`.

Reply:
No way, we're already using it for another key